### ISSUE TITLE
Implement autocomplete jump to identifier on package page

### ISSUE THREAD

**Issue by garyburd** [golang/gddo#128]
If user types "#" in search box on package page, then display autocomplete typeahead for exported identifiers on the page.  When user hits enter with leading "#" in query, then jump to the anchor instead of searching.

http://twitter.github.io/typeahead.js/ will be useful in implementing this feature.

**Comment by koorgoo** [comment]
That's what I miss now. :+1:

**Comment by shurcooL** [comment]
Can this be done in the same way I've implemented jump to identifier on gotools.org? When you press R anywhere on the page, a select-list-view with all identifiers appears. Typing filters/searches through the list.

**Comment by garyburd** [comment]
Maybe.  Where's the code?

**Comment by shurcooL** [comment]
First, take a quick look at this sample of godoc.org/net/http page with the select-list-view component added (focus the page and press R to open it).

[![image](https://cloud.githubusercontent.com/assets/1924134/6114440/03dfd22e-b053-11e4-8bec-1ea2493d8982.png)](https://dl.dropboxusercontent.com/u/8554242/temp/gddo-with-select-list-view-demo/index.html)

https://dl.dropboxusercontent.com/u/8554242/temp/gddo-with-select-list-view-demo/index.html

The only change to the the .html (aside from turning relative paths into absolute paths) is:

```diff
--- original.html
+++ modified.html
@@ -4,6 +4,7 @@
   <meta name="viewport" content="width=device-width, initial-scale=1.0">
   <link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.1/css/bootstrap.min.css" rel="stylesheet">
   <link href="https://godoc.org/-/site.css?v=bf1ccd6efc28f5d787d7c28c6d1c8d97e5ec1b47" rel="stylesheet">
+  <link href="select-list-view-style.css" rel="stylesheet">
 
 
   <title>http - GoDoc</title>
@@ -2663,5 +2664,6 @@
 <script src="https://ajax.googleapis.com/ajax/libs/jquery/2.0.3/jquery.min.js"></script>
 <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.1/js/bootstrap.min.js"></script>
 <script src="https://godoc.org/-/site.js?v=d652d20bb449139d8586d925bdadedf46bdf64c5"></script>
+<script src="select-list-view-script.js"></script>
 </body>
 </html>
```

The reason it works with just .js and .css is because, by default, select-list-view component gets all of its entries by taking all \<h3\> and \<h4\> elements on the page. godoc.org already uses those elements for all the entries on the page. But I would imagine we'd want to tweak things further to achieve an optimal experience.

> Maybe. Where's the code?

The code is at https://github.com/shurcooL/frontend/tree/master/select-list-view. The [README](https://github.com/shurcooL/frontend#frontend) in repo root explains things and shows how to get started. It's written in Go and can be compiled to JavaScript using the current version of GopherJS.

**Comment by garyburd** [comment]
I think it's possible to implement the feature in a few lines of Javascript using existing dependencies.

**Comment by garyburd** [comment]
It is a few lines of Javscript. I'll have something soon.

**Comment by shurcooL** [comment]
I agree, it is possible to implement this feature in a few lines of JavaScript using existing JavaScript dependencies and in a few lines of Go using existing Go dependencies.

**Comment by shurcooL** [comment]
I think this issue has been resolved via #242.
